### PR TITLE
Update EFS CSI driver to include support dynamic provisioning

### PR DESCRIPTION
**Pull request by sebastianlzy**
*Issue #, if available:*

*Description of changes:*

Update with latest support information for EFS CSI driver on support for dynamic provisioning

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by pjlewisuk — APPROVED** [review]
LGTM

File: content/security/docs/data.md [hunk 1]
```diff
@@ -1,7 +1,9 @@
...
+There are three different AWS-native storage options you can use with Kubernetes: [EBS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AmazonEBS.html), [EFS](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AmazonEFS.html), and [FSx for Lustre](https://docs.aws.amazon.com/fsx/latest/LustreGuide/what-is.html).  All three offer encryption at rest using a service managed key or a customer master key (CMK). For EBS you can use the in-tree storage driver or the [EBS CSI driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver).  Both include parameters for encrypting volumes and supplying a CMK.  For EFS, you can use the [EFS CSI driver](https://github.com/kubernetes-sigs/aws-efs-csi-driver). As of 23 March 2021, EFS CSI supports both [static and dynamic provisioning](https://aws.amazon.com/blogs/containers/introducing-efs-csi-dynamic-provisioning/)
```
Comment (jicowan) [inline content/security/docs/data.md:4]: I see you removed the blurb on EFS's lack of support for dynamic provisioning. EFS can dynamically provision Access Points but it is still incapable of dynamically creating file systems. I think we need to be clear about that.

Comment (nublarsec) [reply content/security/docs/data.md:4]: Agree with @jicowan, this seems to be a slight misunderstanding. In the [AWS Blog post you added](https://aws.amazon.com/blogs/containers/introducing-efs-csi-dynamic-provisioning) they assume you have an existing EFS Filesystem provisioned already.

The dynamic provisioning feature creates EFS Access Points in the given filesystem when it detects a Persistent Volume Claim request. These get their own UID:GID and a dedicated directory in the Filesystem (with defaults it looks like `/dynamic_provisioning/pvc-12341231234...`). More details [in the official example](https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/examples/kubernetes/dynamic_provisioning).

I suggest keeping the original language but clarifying this a bit.

Comment (jicowan) [reply content/security/docs/data.md:4]: @sebastianlzy can you revise the PR to make this clearer?

**Comment by svennam92** [comment]
Please re-open this PR against the asciidoc format we've migrated to (folder: latest/bpg)

**Pull request closed without merging**